### PR TITLE
refactor: Create text copy handler

### PR DESCRIPTION
**Pull request by TerryHowe**
**What this PR does / why we need it**:

This is a baby step to creating copy handlers for the cp command

### CODE REVIEW

File: cmd/oras/root/cp.go [hunk 1]
```diff
@@ -179,24 +180,11 @@ func doCopy(ctx context.Context, printer *output.Printer, src oras.ReadOnlyGraph
...
-			if err := output.PrintSuccessorStatus(ctx, desc, dst, committed, printer.StatusPrinter(promptSkipped)); err != nil {
-				return err
-			}
-			return printer.PrintStatus(desc, promptCopied)
-		}
-		extendedCopyOptions.OnMounted = func(ctx context.Context, desc ocispec.Descriptor) error {
-			committed.Store(desc.Digest.String(), desc.Annotations[ocispec.AnnotationTitle])
-			return printer.PrintStatus(desc, promptMounted)
-		}
+		copyHandler := status.NewTextCopyHandler(printer, dst)
```
Comment (qweeah) [inline cmd/oras/root/cp.go:183]: Should merge initialization to `display.NewCopyHandler` and gets both handler for status and metadata output.

File: cmd/oras/internal/display/status/text.go [hunk 2]
```diff
@@ -118,3 +118,42 @@ func NewTextPullHandler(printer *output.Printer) PullHandler {
...
+	}
+}
+
+// OnCopySkipped is called when an object already exists.
+func (ch *TextCopyHandler) OnCopySkipped(_ context.Context, desc ocispec.Descriptor) error {
+	ch.committed.Store(desc.Digest.String(), desc.Annotations[ocispec.AnnotationTitle])
+	return ch.printer.PrintStatus(desc, copyPromptExists)
+}
+
+func (ch *TextCopyHandler) PreCopy(_ context.Context, desc ocispec.Descriptor) error {
```
Comment (qweeah) [inline cmd/oras/internal/display/status/text.go:144]: Missing comment doc.

File: cmd/oras/internal/display/status/text.go [hunk 3]
```diff
@@ -118,3 +118,42 @@ func NewTextPullHandler(printer *output.Printer) PullHandler {
...
+	return ch.printer.PrintStatus(desc, copyPromptCopying)
+}
+
+func (ch *TextCopyHandler) PostCopy(ctx context.Context, desc ocispec.Descriptor) error {
```
Comment (qweeah) [inline cmd/oras/internal/display/status/text.go:148]: Missing comment doc.

File: cmd/oras/internal/display/status/text.go [hunk 4]
```diff
@@ -118,3 +118,42 @@ func NewTextPullHandler(printer *output.Printer) PullHandler {
...
+	ch.committed.Store(desc.Digest.String(), desc.Annotations[ocispec.AnnotationTitle])
+	if err := output.PrintSuccessorStatus(ctx, desc, ch.fetcher, ch.committed, ch.printer.StatusPrinter(copyPromptSkipped)); err != nil {
+		return err
+	}
+	return ch.printer.PrintStatus(desc, copyPromptCopied)
+}
+
+func (ch *TextCopyHandler) OnMounted(ctx context.Context, desc ocispec.Descriptor) error {
```
Comment (qweeah) [inline cmd/oras/internal/display/status/text.go:156]: Missing comment doc.

**Review by qweeah — CHANGES_REQUESTED** [review]
Still need to add TTY implementation of status.CopyHandler. Not necessarily needed in this PR.

File: cmd/oras/root/cp.go (hunk 1 above)
Comment (TerryHowe) [reply cmd/oras/root/cp.go]: Done

**Comment by TerryHowe** [comment]
> Still need to add TTY implementation of status.CopyHandler. Not necessarily needed in this PR.

If this PR merges, it will be easier to create a tty handler and the successor PR is so bloated.

**Review by qweeah — APPROVED** [review]
LGTM

**Pull request merged**